aiida/restapi/translator/nodes/node.py

```diff
@@ -563,7 +563,10 @@ def get_formatted_result(self, label):
 
         for node_entry in results[result_name]:
             # construct full_type and add it to every node
-            node_entry['full_type'] = construct_full_type(node_entry.get('node_type'), node_entry.get('process_type'))
+            node_entry['full_type'] = (
+                construct_full_type(node_entry.get('node_type'), node_entry.get('process_type'))
+                if node_entry.get('node_type') or node_entry.get('process_type') else None
+            )
```

Comment:
Oh, I see, so here is where the `full_type` is set?
Just for me to understand: This is the `NodeTranslator` class, but for some reason it is also used to translate objects that re not nodes - is that the source of the problem?

And, finally, I guess you tried before instead of setting `full_type` to `None` to simply not set it here, but then tests break?

Reply:
Yeah, so this is what I tried to explain in previous comment answers.
The only way to get _all_ the various REST API-specific information in the response (and make it the most similar to other REST API responses), I need to use `NodeTranslator` as the translator class for the `/querybuilder`-endpoint.
This is because the `NodeTranslator` is special, adding `full_type`, which is not a property that otherwise exists in AiiDA. It's a quirk of the REST API.
As far as I know, this is the only addition to an AiiDA entity, and as such, if I use `NodeTranslator` for all entities, I make sure I also include the special REST API properties. The need for the subsequent _removal_ of `full_type` should now be clear - it's not a property of any other AiiDA entity than `Node`s. And even then, to define `full_type` both `node_type` and `process_type` are needed. If these properties are not requested in the POSTed queryhelp, they will not be available.
I could here ensure that they're always requested, but that would demand a lot of logic to go through the POSTed queryhelp. Something I didn't feel was necessary for this PR at this point. So instead I've opted for the current solution.

It's worth noting that the `construct_full_type` utility function actually had a bug. It used `process_type` twice (for both `node_type` and `process_type` parts of `full_type`). This PR fixes that bug, as well as remove the necessity for `try/except` in the highlighted code.

Reply:
> And, finally, I guess you tried before instead of setting `full_type` to `None` to simply not set it here, but then tests break?

It was more of a conscious choice to make the response as AiiDA REST API-like as possible. Including all the extra properties expected from any other AiiDA REST API response for the various entities.

Reply:
The main issue here relates to the fact that the REST API was _not_ built to return multiple types of entities (this has been mentioned in issue #4676 as well). So I need a "catch-em-all"/"works-for-all"-translator :)